### PR TITLE
ref(docs): consolidate docs linting and publishing

### PR DESCRIPTION
**Pull request by gustavovalverde**
## Motivation

We were not linting before publishing

Depends-On: #7338

## Solution

- Consolidate linting errors with the docs build and publish steps

## Review

This should fail if there are  Rust `Warnings` while building the docs. And those should be fixed in this PR

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
    - [ ] Does it change concurrent code, unsafe code, or consensus rules?
  - [ ] How do you know it works? Does it have tests?

## Follow Up Work

None

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -149,25 +154,25 @@ jobs:
...
+      - name: Add $GCP_FIREBASE_SA_PATH to env
+        run: |
           # shellcheck disable=SC2002
-          echo "GCP_FIREBASE_SA=$(cat ${{ steps.auth.outputs.credentials_file_path }} | tr -d '\n')" >> "$GITHUB_ENV"
+          echo "GCP_FIREBASE_SA_PATH=$(cat ${{ steps.auth.outputs.credentials_file_path }} | tr -d '\n')" >> "$GITHUB_ENV"
 
       - name: Deploy external docs to firebase
         uses: FirebaseExtended/action-hosting-deploy@v0.7.1
         with:
           repoToken: ${{ secrets.GITHUB_TOKEN }}
```
Comment (arya2) [inline .github/workflows/docs.yml]: Optional: Could we try removing these repoTokens?

**Review by arya2 — DISMISSED** [review]

File: .github/workflows/docs.yml (hunk 1 above)
Comment (gustavovalverde) [reply .github/workflows/docs.yml]: Fixed with: [fix(docs): remove preview URLs from deployments](https://github.com/ZcashFoundation/zebra/pull/7339/commits/04c41e44c252f2278afbc63185bb6eafe53e14cb)

**Review by arya2 — DISMISSED** [review]
Thank you!

**Comment by gustavovalverde** [comment]
Sorry @arya2, this will sadly need re-approval

**Review by arya2 — APPROVED** [review]
Thank you for this fix.

**Pull request merged**